pkg/sectigo/sectigo_test.go

```diff
@@ -1,21 +1,223 @@
+}
+
+func (s *SectigoTestSuite) TestCredsCopy() {
+	require := s.Require()
 
-	require.Equal(t, api.creds.Username, creds.Username)
+	// Test the internal Sectigo credentials
+	creds := s.api.Creds()
+	require.Equal("foo", creds.Username)
+	require.Equal("supersecret", creds.Password)
+
+	// Ensure that creds are copied and are not the same object
 	creds.Username = "superbunny"
-	require.NotEqual(t, api.creds.Username, creds.Username)
-	require.Equal(t, api.creds.Username, "foo")
+	creds.Password = "knockknock"
+	require.NotEqual(&creds, s.api.Creds())
+
+	orig := s.api.Creds()
+	require.NotEqual(creds.Username, orig.Username)
+	require.NotEqual(creds.Password, orig.Password)
+}
+
+func (s *SectigoTestSuite) TestSuccessfulCalls() {
+	require := s.Require()
+	tests := []struct {
+		name string
+		f    func(t *testing.T)
+	}{
+		{name: AuthenticateEP, f: s.authenticate},
+		{name: RefreshEP, f: s.refresh}, // must be called after authenticate
+		{name: CreateSingleCertBatchEP, f: s.createSingleCertBatch},
+		{name: UploadCSREP, f: s.uploadCSRBatch},
+		{name: BatchDetailEP, f: s.batchDetail},
+		{name: BatchProcessingInfoEP, f: s.processingInfo},
+		{name: DownloadEP, f: s.download},
+		{name: DevicesEP, f: s.licensesUsed},
+		{name: UserAuthoritiesEP, f: s.userAuthorities},
+		{name: AuthorityBalanceAvailableEP, f: s.authorityAvailableBalance},
+		{name: CurrentUserOrganizationEP, f: s.organization},
+		{name: ProfilesEP, f: s.profiles},
+		{name: ProfileParametersEP, f: s.profileParams},
+		{name: ProfileDetailEP, f: s.profileDetail},
+		{name: FindCertificateEP, f: s.findCertificate},
+		{name: RevokeCertificateEP, f: s.revokeCertificate},
+	}
+	for _, t := range tests {
+		s.T().Run(t.name, t.f)
+	}
+	calls := s.server.GetCalls()
+	for endpoint, count := range calls {
+		require.Equal(1, count, fmt.Errorf("wrong number of calls to endpoint %s", endpoint))
+	}
+}
+
+func (s *SectigoTestSuite) authenticate(t *testing.T) {
+	require.NoError(t, s.api.Authenticate())
+}
+
+// Note: refresh has an external dependency on the refresh token being set in the
+// sectigo client Credentials object.
+func (s *SectigoTestSuite) refresh(t *testing.T) {
+	require.NoError(t, s.api.Refresh())
+}
+
+func (s *SectigoTestSuite) createSingleCertBatch(t *testing.T) {
+	rep, err := s.api.CreateSingleCertBatch(42, "foo", map[string]string{"foo": "bar"})
+	require.NoError(t, err)
+	require.NotNil(t, rep)
+}
+
+func (s *SectigoTestSuite) uploadCSRBatch(t *testing.T) {
+	rep, err := s.api.UploadCSRBatch(42, "foo", []byte("foo"), map[string]string{"foo": "bar"})
+	require.NoError(t, err)
+	require.NotNil(t, rep)
+}
+
+func (s *SectigoTestSuite) batchDetail(t *testing.T) {
+	rep, err := s.api.BatchDetail(42)
+	require.NoError(t, err)
+	require.NotNil(t, rep)
+}
+
+func (s *SectigoTestSuite) processingInfo(t *testing.T) {
+	rep, err := s.api.ProcessingInfo(42)
+	require.NoError(t, err)
+	require.NotNil(t, rep)
+}
+
+func (s *SectigoTestSuite) download(t *testing.T) {
```

Comment:
I suppose we will eventually have to modify this test to actually check the download, but we can get to that during integration tests.

Reply:
Thanks for catching this, I've added some lightweight checking for now.